W5500.device.nut

```diff
@@ -0,0 +1,2572 @@
+    // setSocketInterrupt
+    // Returns: this
+    // Parameters:
+    //      socketInt - select the sockets to enable interrupts on using
+    //                  SOCKET INTERRUPTS constants or-ed together
+    //      type(optional) - select SOCKET INTERRUPT TYPES using constants
+    //              or-ed together, if type not passed in all type interrupts
+    //              will be cleared.
+    // **************************************************************************
+    function setSocketInterrupt(socketInt, type = W5500_ALL_INT_TYPES) {
+        writeReg(W5500_SOCKET_INTERRUPT_MASK, W5500_COMMON_REGISTER, socketInt);
+        // default enables all socket interrupt types
+        writeReg(W5500_SOCKET_N_INTERRUPT_MASK, W5500_COMMON_REGISTER, type);
+        return this;
+    }
+
+    // ***************************************************************************
+    // clearSocketInterrupt
+    // Returns: this
+    // Parameters:
+    //      socket : socket (selected by integer) to clear interrupt on
+    //      type(optional) - clear specified interrupt type
+    //                       if nothing passed in clears all interrupts
+    // **************************************************************************
+    function clearSocketInterrupt(socket, type = W5500_ALL_INT_TYPES) {
+        local bsb = _getSocketRegBlockSelectBit(socket);
+        writeReg(W5500_SOCKET_N_INTERRUPT, bsb, type);
+        return this;
+    }
+
+    // ***************************************************************************
+    // clearSocketInterrupts
+    // Returns: clears the Sockets Interrupt
+    // Parameters:
+    //      none
+    // **************************************************************************
+    function clearSocketInterrupts() {
+        for (local socket = 0; socket < _noOfSockets; socket++) {
+            clearSocketInterrupt(socket);
+        }
+    }
+
+
+    // ***************************************************************************
+    // handleInterrupt
+    // Returns: interrupt status table
+    // Parameters: none
+    // **************************************************************************
+    function handleInterrupt() {
+
+        // Handle the main interrupt
+        local status1 = readReg(W5500_INTERRUPT, W5500_COMMON_REGISTER);
+        local interrupt = {
+            "CONFLICT": status1 & W5500_CONFLICT_INT_TYPE ? true : false,
+            "UNREACH": status1 & W5500_UNREACH_INT_TYPE ? true : false,
+            "PPPoE": status1 & W5500_PPPoE_INT_TYPE ? true : false,
+            "MAGIC_PACKET": status1 & W5500_MAGIC_PACKET_TYPE ? true : false,
+            "REGISTER_VALUE": status1
+        };
+
+        if (interrupt.UNREACH) {
+            clearInterrupt(W5500_UNREACH_INT_TYPE);
+            // handleUnreachableInt();
+        }
+        if (interrupt.CONFLICT) {
+            clearInterrupt(W5500_CONFLICT_INT_TYPE);
+            handleConflictInt();
```

Comment:
Current function doesn't do anything.  may want to do some clean up here before releasing library.